ndb/src/google/cloud/ndb/model.py

```diff
@@ -336,6 +339,124 @@ def __hash__(self):
 
 
 class Property(ModelAttribute):
+    """A class describing a typed, persisted attribute of an entity.
+
+    .. warning::
+
+        This is not to be confused with Python's ``@property`` built-in.
+
+    .. note::
+
+        This is just a base class; there are specific subclasses that
+        describe properties of various types (and :class:`GenericProperty`
+        which describes a dynamically typed property).
+
+    All special property attributes, even those considered "public",
+    have names starting with an underscore, because :class:`StructuredProperty`
+    uses the non-underscore attribute namespace to refer to nested
+    property names; this is essential for specifying queries on
+    subproperties.
+
+    The :class:`Property` class and its predefined subclasses allow easy
+    subclassing using composable (or stackable) validation and
+    conversion APIs. These require some terminology definitions:
+
+    * A **user value** is a value such as would be set and accessed by the
+      application code using standard attributes on the entity.
+    * A **base value** is a value such as would be serialized to
+      and deserialized from Cloud Datastore.
+
+    A property will be a member of a :class:`Model` and will be used to help
+    store values in an ``entity`` (i.e. instance of a model subclass). The
+    value(s) for a given property ``name`` will be stored in
+    ``entity._values[name]`` and accessed by :meth:`_store_value` and
+    :meth:`_retrieve_value`. For these methods, the values can be either user
+    values or base values. To retrieve user values, use
+    :meth:`_get_user_value`. To retrieve base values, use
+    :meth:`_get_base_value`. In particular, :meth:`_get_value` calls
+    :meth:`_get_user_value`, and :meth:`_serialize` effectively calls
+    :meth:`_get_base_value`.
+
+    To store a user value, just call :meth:`_store_value`. To store a
+    base value, wrap the value in a :class:`_BaseValue` and then
+    call :meth:`_store_value`.
```

Comment:
Why not `_store_user_value` and `_store_base_value`?

Reply:
Good question. It's a house of cards I suppose. The whole `_BaseValue` concept perplexes me.

Reply:
It's not as important to me if we're not going to pawn it off on the users. Not that the code shouldn't be clean internally, but if we're taking these out of the "public" API, then it's just an implementation detail.